test-websocket/app/frontend/index.js

```diff
@@ -0,0 +1,169 @@
+                return;
+            }
+
+            console.log("FROM RAILS: ", msg);
+
+            // Renders any newly created messages onto the page
+            if (msg.message) {
+                renderMessage(msg.message)
+            }
+
+          };
+
+        // When an error occurs through the websocket connection, this code is run printing the error message
+        socket.onerror = function(error) {
+            console.log('WebSocket Error: ' + error);
+        };
+}
+
+document.addEventListener('DOMContentLoaded',() => {
+    // Fetch & render all chat rooms from the server
+    fetch(`${apiUrl}/chat_rooms`)
+        .then(response => response.json())
+        .then(chatRoomsObject => {
+            renderChatRoomsList(chatRoomsObject);
+        })
+
+    // Allow users to create new chat rooms
+    const newChatRoomForm = document.getElementById('new-chat-room-form');
+    newChatRoomForm.addEventListener('submit', event => {
+        event.preventDefault();
+        fetch(`${apiUrl}/chat_rooms`,{
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'Accept': 'application/json'
+            },
+            body: JSON.stringify({
+                name: event.target[0].value
+            })
+        }).then(response => response.json())
+        .then(chatRoomObject => {renderChatRoomOnList(chatRoomObject)})
+
+        newChatRoomForm.reset();
+    })
+
+    // Allow users to join new chat rooms
+    chatRoomListDiv.addEventListener('click', event => {
+        if (event.target.className === 'join-chat-room-button') {
+
+            // Hide the chat room list
+            const chatRoomsDiv = document.getElementById('chat-rooms-div')
+            chatRoomsDiv.style.display = 'none'
+
+            // Fetch & render info on a specific chat room from the server
+            fetch(`${apiUrl}/chat_rooms/${event.target.dataset.chatRoomId}`)
+                .then(response => response.json())
+                .then(chatRoomObject => {
+                    renderChatRoom(chatRoomObject)
+                })
+
+            // Open up a websocket connection for that specific chat room
+            createChatRoomWebsocketConnection(event.target.dataset.chatRoomId)
+        }
+    })
+
+    // Allow users to submit new messages
+    newMessageForm.addEventListener('submit', event => {
+        event.preventDefault();
+        fetch(`${apiUrl}/messages`,{
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'Accept': 'application/json'
+            },
+            body: JSON.stringify({
+                content: event.target[0].value,
+                chat_room_id: event.target.dataset.chatRoomId
+            })
+        })
+        // .then(response => response.json())
+        // .then(messageObject => {renderMessage(messageObject)})
+
+        newMessageForm.reset();
+    })
+})
```

Comment:
I think there are some issues with the index.js. When the browser loads, it keeps refreshing non stop.

Reply:
![problem](https://user-images.githubusercontent.com/67599130/127766006-cefb9a53-f046-4e6b-83c4-485e18ff42a0.gif)